x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferTrainedModelDeploymentAction.java

```diff
@@ -139,7 +162,7 @@ public Request(StreamInput in) throws IOException {
                 skipQueue = in.readBoolean();
             }
             if (in.getVersion().onOrAfter(Version.V_8_7_0)) {
-                textInput = in.readOptionalString();
+                textInput = in.readOptionalStringList();
             } else {
                 textInput = null;
```

Comment:
I think we should read `in.readOptionalString()` and set it as a single item list on `textInput` for older versions.

Reply:
I'm modifying something that hasn't been released yet so it is ok to change it here. I will have to disable the BWC tests until the change is backported to 8.7 but after the backport it will be fine.

I added `textIinput` as a string to support for semantic searc. I changed it to a list here  to be consistent with using a  list of docs

Reply:
Right, yes, I hadn't realised `textInput` was added in 8.7.